### PR TITLE
Add subscribe event type for donations queue

### LINKED ISSUE #152: adding a new event type to basket
For the foundation website's newsletter(s) signup functionality, we'd like to send signup events on a new event type. We're currently using `crm_petition_data` for mofo petitions, but we'd like to also have a `newsletter_signup_data` event type for dealing with (comma-separated) newsletter(s) that users can register for.

### PR DESCRIPTION
**Pull request by pmac**
Fix #152

### CODE REVIEW

**Comment by pmac** [comment]
@Pomax FYI

File: basket/news/tasks.py [hunk 1]
```diff
@@ -928,6 +928,12 @@ def process_donation(data):
             raise
 
 
+@et_task
+def process_newsletter_subscribe(data):
+    data['lang'] = get_best_supported_lang(data['lang'])
```
Comment (jpetto) [inline basket/news/tasks.py]: Is there any need to ensure `lang` exists in `data`? Or that `data` contains all info needed to `upsert_user`?

File: basket/news/management/commands/process_donations_queue.py [hunk 2]
```diff
@@ -15,13 +15,15 @@
...
     process_donation_event,
+    process_newsletter_subscribe,
     process_petition_signature,
 )
 
 
 EVENT_TYPES = {
     'donation': process_donation,
     'crm_petition_data': process_petition_signature,
+    'newsletter_signup_data': process_newsletter_subscribe,
```
Comment (jpetto) [inline basket/news/management/commands/process_donations_queue.py]: Out of curiosity, where are these used?

File: basket/news/tasks.py (hunk 1 above)
Comment (pmac) [reply basket/news/tasks.py]: It should all always be there. See the code on the mofo side:

https://github.com/mozilla/foundation.mozilla.org/blob/060922af4dd07f35f8119f2596228f47946b4487/network-api/networkapi/campaign/views.py#L87-L93

I really only did the `best_supported_lang` call to prepare for them someday sending a real language. It's hard coded to `en` for now.

File: basket/news/management/commands/process_donations_queue.py (hunk 2 above)
Comment (pmac) [reply basket/news/management/commands/process_donations_queue.py]: What are the `these` in this question? These `EVENT_TYPES`? If so they're used just a few lines below this in this file.

**Comment by pmac** [comment]
The details can be found in the linked issue (#152) where the code on the foundation's site can be found. I linked it above.

File: basket/news/management/commands/process_donations_queue.py (hunk 2 above)
Comment (jpetto) [reply basket/news/management/commands/process_donations_queue.py]: Ah, I see. I did a search for the types and only found `donation`. The other types are specified in the incoming data. Got it.

**Comment by jpetto** [comment]
All good. Apologies for not reading the related issue fully. :shipit: !!

**Comment by pmac** [comment]
@jpetto oh no need for apology. I should have pointed all of that out in the PR description. Thanks for being thorough!

**Pull request merged**